{
  "active_lock_reason": null,
  "assignee": {
    "avatar_url": "https://avatars2.githubusercontent.com/u/676637?v=4",
    "events_url": "https://api.github.com/users/feiskyer/events{/privacy}",
    "followers_url": "https://api.github.com/users/feiskyer/followers",
    "following_url": "https://api.github.com/users/feiskyer/following{/other_user}",
    "gists_url": "https://api.github.com/users/feiskyer/gists{/gist_id}",
    "gravatar_id": "",
    "html_url": "https://github.com/feiskyer",
    "id": 676637,
    "login": "feiskyer",
    "node_id": "MDQ6VXNlcjY3NjYzNw==",
    "organizations_url": "https://api.github.com/users/feiskyer/orgs",
    "received_events_url": "https://api.github.com/users/feiskyer/received_events",
    "repos_url": "https://api.github.com/users/feiskyer/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/feiskyer/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/feiskyer/subscriptions",
    "type": "User",
    "url": "https://api.github.com/users/feiskyer"
  },
  "assignees": [
    {
      "avatar_url": "https://avatars2.githubusercontent.com/u/676637?v=4",
      "events_url": "https://api.github.com/users/feiskyer/events{/privacy}",
      "followers_url": "https://api.github.com/users/feiskyer/followers",
      "following_url": "https://api.github.com/users/feiskyer/following{/other_user}",
      "gists_url": "https://api.github.com/users/feiskyer/gists{/gist_id}",
      "gravatar_id": "",
      "html_url": "https://github.com/feiskyer",
      "id": 676637,
      "login": "feiskyer",
      "node_id": "MDQ6VXNlcjY3NjYzNw==",
      "organizations_url": "https://api.github.com/users/feiskyer/orgs",
      "received_events_url": "https://api.github.com/users/feiskyer/received_events",
      "repos_url": "https://api.github.com/users/feiskyer/repos",
      "site_admin": false,
      "starred_url": "https://api.github.com/users/feiskyer/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/feiskyer/subscriptions",
      "type": "User",
      "url": "https://api.github.com/users/feiskyer"
    }
  ],
  "author_association": "MEMBER",
  "body": "<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!\r\n\r\nIf the matter is security related, please disclose it privately via https://kubernetes.io/security/\r\n-->\r\n\r\n\r\n**What happened**:\r\n\r\nkube-controller-manager panic when network interfaces for Azure VMSS are null. This happens when Azure cloud provider tried to get IP address for VMSS VM:\r\n\r\n```\r\nI0823 22:21:23.235856       1 route_controller.go:176] Creating route for node aks-node-1 10.244.7.0/24 with hint b5759222-0000-45dd-9ecd-11110683608f, throttled 800ns\r\npanic: runtime error: invalid memory address or nil pointer dereference\r\n[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x22335bd]\r\n \r\ngoroutine 257858 [running]:\r\nk8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure.(*scaleSet).getPrimaryInterfaceID(0xc000e2bcb0, 0x0, 0xc0053a97d0, 0xc001236d80, 0xc0048ebd00, 0x0, 0xc0012371e0, 0x0, 0xc0053a97e0, 0xc0053a96d0, ...)\r\n        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure/azure_vmss.go:443 +0x2d\r\nk8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure.(*scaleSet).GetPrimaryInterface(0xc000e2bcb0, 0xc004617b90, 0x23, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)\r\n        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure/azure_vmss.go:657 +0x52e\r\nk8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure.(*scaleSet).GetIPByNodeName(0xc000e2bcb0, 0xc004617b90, 0x23, 0xc005a66480, 0x30, 0x30, 0x509e9e0, 0xc000069470, 0xc000069400)\r\n        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure/azure_vmss.go:382 +0x80\r\nk8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure.(*Cloud).GetIPForMachineWithRetry.func1(0x509e9e0, 0xc003fab901, 0xc005a66480)\r\n        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure/azure_backoff.go:139 +0x79\r\nk8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff(0x0, 0x0, 0x0, 0x1, 0x0, 0xc005a66480, 0x0, 0x0)\r\n        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:284 +0x51\r\nk8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure.(*Cloud).GetIPForMachineWithRetry(0xc0009f6f00, 0xc004617b90, 0x23, 0x5333ec0, 0xc005a66450, 0xc004617b90, 0x23, 0xc375220, 0x433200)\r\n        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure/azure_backoff.go:137 +0x13e\r\nk8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure.(*Cloud).getIPForMachine(0xc0009f6f00, 0xc004617b90, 0x23, 0x443700, 0x0, 0x0, 0xc003fabad8, 0x443bed, 0xc005ba5ec0)\r\n        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure/azure_backoff.go:131 +0xe3\r\nk8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure.(*Cloud).CreateRoute(0xc0009f6f00, 0x88d54e0, 0xc00005c050, 0x5516421, 0xa, 0xc004617bc0, 0x24, 0xc003bedb00, 0x0, 0x0)\r\n        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/azure/azure_routes.go:286 +0x2b8\r\nk8s.io/kubernetes/pkg/controller/route.(*RouteController).reconcile.func1.1(0x7fb7008f0b28, 0x0)\r\n        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/route/route_controller.go:177 +0x2d4\r\nk8s.io/kubernetes/vendor/k8s.io/client-go/util/retry.RetryOnConflict.func1(0x4e76a20, 0xc003b86601, 0xc0046ffc20)\r\n        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/util/retry/util.go:64 +0x33\r\nk8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff(0x5f5e100, 0x0, 0x3ff0000000000000, 0x5, 0x0, 0xc0046ffc20, 0x40fed8, 0x40)\r\n        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:284 +0x51\r\nk8s.io/kubernetes/vendor/k8s.io/client-go/util/retry.RetryOnConflict(0x5f5e100, 0x0, 0x3ff0000000000000, 0x5, 0x0, 0xc003bedb40, 0x1, 0xc00426dcb0)\r\n        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/util/retry/util.go:63 +0xa6\r\nk8s.io/kubernetes/pkg/controller/route.(*RouteController).reconcile.func1(0xc002f94bd0, 0xc002219d40, 0xc0003d41c0, 0xc004617b90, 0x23, 0xc004617bc0, 0x24, 0xc003bedb00)\r\n        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/route/route_controller.go:171 +0x123\r\ncreated by k8s.io/kubernetes/pkg/controller/route.(*RouteController).reconcile\r\n        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/route/route_controller.go:169 +0x38c\r\npanic: runtime error: invalid memory address or nil pointer dereference\r\n[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x22335bd]\r\n```\r\n\r\n**What you expected to happen**:\r\n\r\nkube-controller-manager should report an error instead of panic.\r\n\r\n**How to reproduce it (as minimally and precisely as possible)**:\r\n\r\nScale cluster nodes up and down frequently.\r\n\r\n**Anything else we need to know?**:\r\n\r\n/kind bug\r\n/area provider/azure\r\n\r\n**Environment**:\r\n- Kubernetes version (use `kubectl version`): 1.15+\r\n- Cloud provider or hardware configuration: Azure\r\n- OS (e.g: `cat /etc/os-release`):\r\n- Kernel (e.g. `uname -a`):\r\n- Install tools:\r\n- Network plugin and version (if this is a network-related bug):\r\n- Others:\r\n",
  "closed_at": "2020-09-02T02:41:44Z",
  "closed_by": {
    "avatar_url": "https://avatars0.githubusercontent.com/u/20407524?v=4",
    "events_url": "https://api.github.com/users/k8s-ci-robot/events{/privacy}",
    "followers_url": "https://api.github.com/users/k8s-ci-robot/followers",
    "following_url": "https://api.github.com/users/k8s-ci-robot/following{/other_user}",
    "gists_url": "https://api.github.com/users/k8s-ci-robot/gists{/gist_id}",
    "gravatar_id": "",
    "html_url": "https://github.com/k8s-ci-robot",
    "id": 20407524,
    "login": "k8s-ci-robot",
    "node_id": "MDQ6VXNlcjIwNDA3NTI0",
    "organizations_url": "https://api.github.com/users/k8s-ci-robot/orgs",
    "received_events_url": "https://api.github.com/users/k8s-ci-robot/received_events",
    "repos_url": "https://api.github.com/users/k8s-ci-robot/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/k8s-ci-robot/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/k8s-ci-robot/subscriptions",
    "type": "User",
    "url": "https://api.github.com/users/k8s-ci-robot"
  },
  "comments": 1,
  "comments_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/94354/comments",
  "created_at": "2020-08-31T08:07:05Z",
  "events_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/94354/events",
  "html_url": "https://github.com/kubernetes/kubernetes/issues/94354",
  "id": 689009417,
  "labels": [
    {
      "color": "0052cc",
      "default": false,
      "description": "Issues or PRs related to azure provider",
      "id": 852130786,
      "name": "area/provider/azure",
      "node_id": "MDU6TGFiZWw4NTIxMzA3ODY=",
      "url": "https://api.github.com/repos/kubernetes/kubernetes/labels/area/provider/azure"
    },
    {
      "color": "e11d21",
      "default": false,
      "description": "Categorizes issue or PR as related to a bug.",
      "id": 105146071,
      "name": "kind/bug",
      "node_id": "MDU6TGFiZWwxMDUxNDYwNzE=",
      "url": "https://api.github.com/repos/kubernetes/kubernetes/labels/kind/bug"
    },
    {
      "color": "d2b48c",
      "default": false,
      "description": "Categorizes an issue or PR as relevant to SIG Cloud Provider.",
      "id": 958178286,
      "name": "sig/cloud-provider",
      "node_id": "MDU6TGFiZWw5NTgxNzgyODY=",
      "url": "https://api.github.com/repos/kubernetes/kubernetes/labels/sig/cloud-provider"
    }
  ],
  "labels_url": "https://api.github.com/repos/kubernetes/kubernetes/issues/94354/labels{/name}",
  "locked": false,
  "milestone": null,
  "node_id": "MDU6SXNzdWU2ODkwMDk0MTc=",
  "number": 94354,
  "performed_via_github_app": null,
  "repository_url": "https://api.github.com/repos/kubernetes/kubernetes",
  "state": "closed",
  "title": "kube-controller-manager panic when network interfaces for Azure VMSS are null",
  "updated_at": "2020-09-02T02:41:44Z",
  "url": "https://api.github.com/repos/kubernetes/kubernetes/issues/94354",
  "user": {
    "avatar_url": "https://avatars2.githubusercontent.com/u/676637?v=4",
    "events_url": "https://api.github.com/users/feiskyer/events{/privacy}",
    "followers_url": "https://api.github.com/users/feiskyer/followers",
    "following_url": "https://api.github.com/users/feiskyer/following{/other_user}",
    "gists_url": "https://api.github.com/users/feiskyer/gists{/gist_id}",
    "gravatar_id": "",
    "html_url": "https://github.com/feiskyer",
    "id": 676637,
    "login": "feiskyer",
    "node_id": "MDQ6VXNlcjY3NjYzNw==",
    "organizations_url": "https://api.github.com/users/feiskyer/orgs",
    "received_events_url": "https://api.github.com/users/feiskyer/received_events",
    "repos_url": "https://api.github.com/users/feiskyer/repos",
    "site_admin": false,
    "starred_url": "https://api.github.com/users/feiskyer/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/feiskyer/subscriptions",
    "type": "User",
    "url": "https://api.github.com/users/feiskyer"
  }
}